### DISCUSSION TITLE
Lesson 6: Why cannot I use .wait(1) on fund_me = FundMe.deploy(...)

### DISCUSSION THREAD

**Discussion by n4n0b1t3** [smartcontractkit/full-blockchain-solidity-course-py#1139] [Q&A] [answered]
**new problem description**
I am an idiot! The mock was excluded because I did not run on a local chain. Thank you @cromewar for making me aware of it.

**problem description**
When deploying to Infura or Alchemy often a `ConnectionResetError` is thrown right after contract deployment. I am trying to fix that with a transaction.wait(1). This works without issues on MockV3Aggregator.deploy() 
```python
    tx = MockV3Aggregator.deploy(DECIMALS, INITIAL_VALUE, {"from": account})
    tx.wait(1)
```

Why, when doing the same with my contract deployment, this gives me the following error?
 `AttributeError: Contract 'FundMe' object has no attribute 'wait'`
Here the code:
```python
    fund_me = FundMe.deploy(
        price_feed_address,
        {"from": account},
        publish_source=config["networks"][network.show_active()].get("verify"),
    )
    fund_me.wait(1)
```
_disclaimer: the code runs fine otherwise, it's just the line fund_me.wait(1) which throws an error._

**environment overview**
- win10/64
- Python 3.9.10
- Brownie v1.18.1

**shortened terminal output with errors**
```python
(py39brownie) fund_me>brownie run .\scripts\deploy.py --network mainnet-fork-infura
[...]
Launching 'ganache-cli.cmd --chain.vmErrorsOnRPCResponse true --wallet.totalAccounts 10 --fork.url https://mainnet.infura.io/v3/my_id --miner.blockGasLimit 12000000 --wallet.mnemonic brownie --server.port 8545 --hardfork istanbul'...

Running 'scripts\deploy.py::main'...
Transaction sent: 0xsometxnumber
  FundMe.constructor confirmed   Block: 14262161   Gas used: 515622 (4.30%)

  File "C:\Users\n4n0b1t3\.virtualEnvironments\py39brownie\lib\site-packages\brownie\_cli\run.py", line 51, in main
    return_value, frame = run(
  File "C:\Users\n4n0b1t3\.virtualEnvironments\py39brownie\lib\site-packages\brownie\project\scripts.py", line 103, in run
    return_value = f_locals[method_name](*args, **kwargs)
  File ".\scripts\deploy.py", line 30, in main
    deploy_fund_me()
  File ".\scripts\deploy.py", line 24, in deploy_fund_me
    fund_me.wait(1)
  File "C:\Users\n4n0b1t3\.virtualEnvironments\py39brownie\lib\site-packages\brownie\network\contract.py", line 765, in __getattribute__
    raise AttributeError(f"Contract '{self._name}' object has no attribute '{name}'")
AttributeError: Contract 'FundMe' object has no attribute 'wait'
Terminating local RPC client...
(py39brownie) fund_me>
```

**Comment by cromewar** [comment, marked as answer]
Hello @n4n0b1t3 
Are you sure this works?
```Python
  tx = MockV3Aggregator.deploy(DECIMALS, INITIAL_VALUE, {"from": account})
    tx.wait(1)
```
I have the same error:

![image](https://user-images.githubusercontent.com/20199180/155336987-17981040-9415-4892-899f-b77b38284f20.png)


And this is because contract deployment are not transactions you would have to wait block  confirmations for, at least not in brownie logic, as the transaction receipt is the actual contract address.

**Reply by n4n0b1t3** [reply]
OK, finally I found the problem: I am an idiot! :-)
I am running this not on local and the deployment of the mock is OF COURSE excluded from external networks.

What would be your approach to avoid network timeouts? I am not so sure the timeouts setting per network makes a difference.

**Comment by ccorderop** [comment]
Hey friends. I'm writting from Spain ..

I have a similar problem ..   i would like ask you if you could give me and idea what happen ..   my error in lesson 6 is:

_raise AttributeError(f"Contract '{self._name}' object has no attribute '{name}'")
AttributeError: Contract 'FundMe' object has no attribute 'getEntranceFee'_


**Script**:
from brownie import FundMe, accounts
from scripts.helpful_scripts import get_account

def fund():
    fund_me = FundMe[-1]
    account = get_account()
    print(account)
    entrance_fee = fund_me.getEntranceFee()
    print(entrance_fee)

def main():
    fund()

**FundMe.sol**:
    _function getEntranceFee() public view returns (uint256) {
        // minimumUSD
        uint256 minimumUSD = 50 * 10 ** 18;
        uint256 price = getPrice();
        uint256 precision = 1 * 10 ** 18;
        // return (minimumUSD * precision) / price;
        // We fixed a rounding error found in the video by adding one!
        return ((minimumUSD * precision) / price) + 1;
    }_

I compiled FundMe.sol and i see in .json getEntranceFee() ..   On ganache instance i see:

_web3_clientVersion
eth_getBlockByNumber
eth_accounts
eth_getCode
eth_getCode_

An Idea what could happen??

Tnx very much and have a nice day.
